numba/runtime/context.py

```diff
@@ -105,6 +105,21 @@ def meminfo_new_varsize(self, builder, size):
         fn.return_value.add_attribute("noalias")
         return builder.call(fn, [size])
 
+
+    def meminfo_new_varsize_dtor(self, builder, size, dtor):
+        """
+        Like meminfo_new_varsize() but also set the destructor for
+        clearing objects inside the allocation.
```

Comment:
clearing? Should this be deallocating?

Reply:
Technically, the destructor is not deallocating the objects inside because they may have reference from elsewhere.  But clearing doesn't sound right either.  I will think of sth better.